### ISSUE TITLE
sft training problem

### ISSUE THREAD

**Issue by sod1010** [Neph0s/CoSER#11]
Hello, thank you for providing the training dataset. I would like to know, for multi-turn dialogue data, what are the input and output formats during SFT training?
Is the input composed of the previous dialogue turns, and the output the current turn?

**Comment by Neph0s** [comment]
Hi, our sft data file follows the sharegpt format. 
In this format, each sample is a list of messages, such as: 
`
[
{"from":"user", "content": "input content"},
{"from":"assistant", "content": "output content for training"},
{"from":"user", "content": "input content2"},
{"from":"assistant", "content": "output content for training2"}
]
`

Hence, we feed the entire multi-turn dialogue to LLMs, and the trainner automatically identifies the "assistant" part that we want to use to train LLMs , and calculate and optimize the loss. 

In short, if the dialog is composed of ['role_1 message1', 'role_2 message1', 'role_1 message2', 'role_2 message2'], and we want to train LLMs as role_2, the trainer simultaneously trains LLMs to predict 'role_2 message1' given ['role_1 message1'], and to predict 'role_2 message2' given ['role_1 message1', 'role_2 message1', 'role_1 message2']

**Comment by sod1010** [comment]
> Hi, our sft data file follows the sharegpt format. In this format, each sample is a list of messages, such as: `[ {"from":"user", "content": "input content"}, {"from":"assistant", "content": "output content for training"}, {"from":"user", "content": "input content2"}, {"from":"assistant", "content": "output content for training2"} ]`
> 
> Hence, we feed the entire multi-turn dialogue to LLMs, and the trainner automatically identifies the "assistant" part that we want to use to train LLMs , and calculate and optimize the loss.
> 
> In short, if the dialog is composed of ['role_1 message1', 'role_2 message1', 'role_1 message2', 'role_2 message2'], and we want to train LLMs as role_2, the trainer simultaneously trains LLMs to predict 'role_2 message1' given ['role_1 message1'], and to predict 'role_2 message2' given ['role_1 message1', 'role_2 message1', 'role_1 message2']

Thank you for your prompt reply, and this paper is very helpful to me!
